avogadro/qtgui/meshgenerator.h

```diff
@@ -81,11 +81,77 @@ class AVOGADROQTGUI_EXPORT MeshGenerator : public QThread
    */
   void run() override;
 
+  /**
+  * It holds the range and starting offsets of isosurface-intersected 
+  * edges along the x, y, and z axes for each grid cell.
+  */
+    struct gridEdge
+    {
+        gridEdge()
+          : xl(0),
+            xr(0),
+            xstart(0),
+            ystart(0),
+            zstart(0)
+        {}
+
+        // trim values
+        // set on pass 1
+        int xl;
+        int xr;
+
+        // modified on pass 2
+        // set on pass 3
+        int xstart;
+        int ystart;
+        int zstart;
+    };
+
+
```

Comment:
To help with formatting the documentation, please add this to group the methods.

```
 /**
   * @name Flying Edges
   * Methods to implement the "flying edges" method for isosurface mesh generation.
   * Flying edges: A high-performance scalable isocontouring algorithm
   * Schroeder; Maynard; Geveci; 
   * 2015 IEEE 5th Symposium on Large Data Analysis and Visualization (LDAV)
   * [10.1109/LDAV.2015.7348069](https://doi.org/10.1109/LDAV.2015.7348069)
   * Alternate (non-VTK) implementation at
   * https://github.com/sandialabs/miniIsosurface/blob/master/flyingEdges/
   * @{
```